File: Core/Interfaces/IOrderRepository.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Core.Models;

namespace Core.Interfaces
{
    public interface IOrderRepository
    {
        Task<Order> GetOne(int id);
        Task<IEnumerable<Order>> GetMany(int pageSize, int offset);
        Task<Order> Store(Order newOrder);
        Task Delete(int id);
    }
}

File: AdminExample/Scripts/interfaces/IOrderRepository.ts
module ShopAdmin {
    export interface IOrderRepository {
        getMany(serquenceLength: number, offset: number): ng.IPromise<Order[]>;
        getOne(id: number): ng.IPromise<Order>;
        store(order: Order): ng.IPromise<Order>;
        deleteOne(id: number): ng.IPromise<any>;
    }
}